safe_transaction_service/history/services/safe_service.py

```diff
@@ -93,23 +98,26 @@ def get_safe_creation_info(self, safe_address: str) -> Optional[SafeCreationInfo
 
             created_time = creation_ethereum_tx.block.timestamp
 
+            # Get the parent trace for the creation
+            # For L2s, `ProxyCreation` event is used to emulate the trace
             parent_internal_tx = self._get_parent_internal_tx(creation_internal_tx)
 
             creator = (parent_internal_tx or creation_ethereum_tx)._from
             proxy_factory = creation_internal_tx._from
 
             master_copy: Optional[str] = None
             setup_data: Optional[bytes] = None
-            data = (
-                bytes(parent_internal_tx.data)
-                if parent_internal_tx
-                else bytes(creation_ethereum_tx.data)
-            )
-            result = self._decode_proxy_factory(data) or self._decode_cpk_proxy_factory(
-                data
-            )
-            if result:
-                master_copy, setup_data = result
+            data_tx = parent_internal_tx if parent_internal_tx else creation_internal_tx
+
+            # A regular ether transfer could trigger a Safe deployment, so it's not guaranteed that there will be
+            # ``data`` for the transaction
+            if data_tx.data:
+                data = bytes(data_tx.data)
+                result = self._decode_proxy_factory(
+                    data
+                ) or self._decode_cpk_proxy_factory(data)
```

Comment:
Can you add documentation for this function? `_decode_cpk_proxy_factory`

Reply:
Done